lib-impl/src/main/java/org/ergoplatform/appkit/impl/UnsignedTransactionBuilderImpl.scala

```diff
@@ -0,0 +1,133 @@
+package org.ergoplatform.appkit.impl
+
+import org.ergoplatform._
+import org.ergoplatform.appkit.{Iso, _}
+import org.ergoplatform.wallet.protocol.context.ErgoLikeStateContext
+import org.ergoplatform.wallet.transactions.TransactionBuilder
+import org.ergoplatform.wallet.boxes.{BoxSelector, DefaultBoxSelector}
+
+import special.collection.Coll
+import special.sigma.Header
+import java.util._
+import java.util.stream.Collectors
+
+import org.ergoplatform.appkit.Parameters.MinChangeValue
+import org.ergoplatform.appkit.Parameters.MinFee
+import scorex.crypto.authds.ADDigest
+
+import scala.collection.JavaConversions
+import JavaHelpers._
+
+class UnsignedTransactionBuilderImpl(val _ctx: BlockchainContextImpl) extends UnsignedTransactionBuilder {
+  private[impl] val _inputs: List[UnsignedInput] = new ArrayList[UnsignedInput]
+  private[impl] val _dataInputs: List[DataInput] = new ArrayList[DataInput]
+  private[impl] var _outputCandidates: List[ErgoBoxCandidate] = new ArrayList[ErgoBoxCandidate]
+  private var _inputBoxes: List[InputBoxImpl] = null
+  private var _dataInputBoxes: List[InputBoxImpl] = new ArrayList[InputBoxImpl]
+  private val _tokensToBurn: List[ErgoToken] = new ArrayList[ErgoToken]
+  private var _feeAmount: Option[Long] = None
+  private var _changeAddress: ErgoAddress = null
+  private var _ph: PreHeaderImpl = null
+
+  override def preHeader(ph: PreHeader): UnsignedTransactionBuilder = {
+    require(_ph == null, "PreHeader is already specified")
+    _ph = ph.asInstanceOf[PreHeaderImpl]
+    this
+  }
+
+  override def boxesToSpend(inputBoxes: List[InputBox]): UnsignedTransactionBuilder = {
+    val items = inputBoxes
```

Comment:
Wouldn't it be better to have null check for _inputBoxes as we do for preHeader?

Reply:
fixed